portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/AnalyticsCloudIndividualTests.testcase

```diff
@@ -0,0 +1,80 @@
+@component-name = "portal-analytics-cloud"
+definition {
+
+	property analytics.cloud.enabled = "true";
+	property minimum.slave.ram = "32";
+	property portal.release = "false";
+	property portal.upstream = "true";
+	property testray.main.component.name = "Analytics Cloud";
+
+	setUp {
+		var analyticsCloudURL = PropsUtil.get("analytics.cloud.url");
+		Navigator.openURL(baseURL = "${analyticsCloudURL}");
+
+		ACLogin.loginAs();
+
+		ACWorkspace.openWorkspace();
+
+		ACSettings.takeTokenFromAnalyticsCloud();
+
+		TestCase.setUpPortalInstance();
+
+		Navigator.openURL();
+
+		User.firstLoginUI();
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		Navigator.openURL();
+
+		ACDXPSettings.disconnectDXPFromAnalyticsCloud();
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "Validate if the user can connect AC to DXP"
+	@priority = "5"
+	test CanSetCustomRangeKnownIndividuals {
+		Navigator.openURL();
+		ACUtils.addSiteAndPage();
+		ACUtils.createDMAndAddToPage();
+		ACDXPSettings.connectDXPtoAnalyticsCloud(siteName = "Site Name");
+
+		for (var n : list "1,2,3,4") {
+			Navigator.gotoSitePage(
+				pageName = "AC Page",
+				siteName = "Site Name");
+
+			DMNavigator.gotoDocumentCP(dmDocumentTitle = "DM AC Title");
+		}
+
+		Pause(locator1 = "10000");
+
+		ACUtils.closeAllSessions();
+
+		Pause(locator1 = "30000");
+
+		ACUtils.launchAC();
+
+		ACProperties.clickPropertyMenuButton();
+
+		ACProperties.clickPropertyMenuItem(propertyName = "Liferay DXP");
+
+		ACNavigation.goToIndividuals();
+		ACIndividualsDashboard.goToKnownIndividualsTab();
+		ACIndividualsDashboard.goToKnownIndividualDetails(individualName = "Test Test");
+		ACTimeFilter.setCustomRange(
+			endDay = "13",
+			endMonth = "July",
+			endYear = "2020",
+			startDay = "1",
+			startMonth = "July",
+			startYear = "2020");
+		ACIndividualsDashboard.assertIndividualsActivitiesTimeRange(timeRange = "July 1, 2020 - July 13, 2020");
```

Comment:
hey @calvinctvo , aside from the time range validation, is there any data on the screen that was changed/updated by this time range change? Maybe it is also important to validate the data being changed.

Reply:
Hi @alexcandidos , I can assert if the date is not equal to my custom range before applying my custom range. Since the date range will always be changing due to the nature of calendar.